File: GUILHERME_DUDU_BERTI/design_patterns/exercicio7/PagamentoAdapter.java
public class PagamentoAdapter implements Pagamento {
    private PagamentoAntigo pagamentoAntigo;

    public PagamentoAdapter(PagamentoAntigo pagamentoAntigo) {
        this.pagamentoAntigo = pagamentoAntigo;
    }

    @Override
    public void pagar(double valor) {
        // Aqui adaptamos a chamada para o método do pagamento antigo
        pagamentoAntigo.realizarPagamento(valor);
    }
}

File: ANDERSON_TESSARO_design_patterns/lib/pagamento_adapter.dart
// Interface moderna
abstract class Pagamento {
  void pagar(double valor);
}

// Classe antiga com método diferente
class PagamentoAntigo {
  void realizarPagamentoAntigo(double quantia) {
    print("Pagamento ANTIGO de R\$ ${quantia.toStringAsFixed(2)} realizado.");
  }
}

// Adapter que implementa a interface moderna e usa a classe antiga
class PagamentoAdapter implements Pagamento {
  final PagamentoAntigo pagamentoAntigo;

  PagamentoAdapter(this.pagamentoAntigo);

  @override
  void pagar(double valor) {
    pagamentoAntigo.realizarPagamentoAntigo(valor);
  }
}
